internal/controller/pkg/resolver/reconciler.go

```diff
@@ -172,6 +182,9 @@ func Setup(mgr ctrl.Manager, o controller.Options) error {
 
 	if o.Features.Enabled(features.EnableAlphaDependencyVersionUpgrades) {
 		opts = append(opts, WithNewDagFn(internaldag.NewUpgradingMapDag))
+		if os.Getenv("AUTOMATIC_DEPENDENCY_DOWNGRADE_ENABLED") == "true" {
+			opts = append(opts, WithDowngradesEnabled())
+		}
```

Comment:
I don't like reading env variables here, but I also didn't want to use `features` since this is not considered as a feature but more like a configuration

Reply:
what about adding it directly to the controller.Options?

Reply:
I don't think it will be a common configuration option and should be visible at that level.

Reply:
but it feels less brittle, this way we risk forgetting updating this if we ever decide to rename the env variable to something else. There is no compile time link between the two places where this env is defined and here.